src/managers/MessageManager.js

```diff
@@ -113,6 +114,91 @@ class MessageManager extends BaseManager {
    * @returns {?Snowflake}
    */
 
+  /**
+   * Edits a message, even if it's not cached.
+   * @param {MessageResolvable} message The message to edit
+   * @param {StringResolvable|APIMessage} [content] The new content for the message
+   * @param {MessageEditOptions|MessageEmbed} [options] The options to provide
+   * @returns {Promise<Message>}
+   */
+  async edit(message, content, options) {
+    message = this.resolveID(message);
+    if (!message) throw new TypeError('INVALID_TYPE', 'message', 'MessageResolvable');
+    const { data } =
+      content instanceof APIMessage ? content.resolveData() : APIMessage.create(this, content, options).resolveData();
+    return this.client.api
+      .channels(this.channel.id)
+      .messages(message)
+      .patch({ data })
+      .then(d => {
+        const existing = this.cache.get(message);
+        if (existing) {
+          const clone = existing._clone();
+          clone._patch(d);
+          return clone;
+        }
+        return this.add(d);
+      });
```

Comment:
Might as well `await` this. 👀 
```suggestion
    const d = await this.client.api
      .channels(this.channel.id)
      .messages(message)
      .patch({ data });
    const existing = this.cache.get(message);
    if (existing) {
      const clone = existing._clone();
      clone._patch(d);
      return clone;
    }
    return this.add(d);
```